association_backend/association/domain/repositories/association_repository.py

```diff
@@ -0,0 +1,118 @@
+import logging
+from typing import List, Optional
+
+import stripe
+from association.domain.entities.stripe_entities import (
+    StripeCheckoutSession,
+    StripeCheckoutSessionInput,
+    StripeCustomer,
+)
+from association.domain.entities.subscription_entities import Subscription
+from association.domain.exceptions import StripeCheckoutSessionNotCreated
+from association.domain.repositories.base import AbstractRepository
+from association.settings import (
+    STRIPE_SUBSCRIPTION_CANCEL_URL,
+    STRIPE_SUBSCRIPTION_PRICE_ID,
+    STRIPE_SUBSCRIPTION_SUCCESS_URL,
+)
+from sqlalchemy import select
+
+logger = logging.getLogger(__name__)
+
+
+class AssociationRepository(AbstractRepository):
+
+    # READ
+    async def get_subscription_by_stripe_id(
+        self, stripe_id: str
+    ) -> Optional[Subscription]:
+        query = select(Subscription).where(Subscription.stripe_id == stripe_id)
+        subscription = (await self.session.execute(query)).scalar_one_or_none()
+        return subscription
+
+    async def get_subscription_by_session_id(
+        self, session_id: str
+    ) -> Optional[Subscription]:
+        query = select(Subscription).where(Subscription.stripe_session_id == session_id)
+        subscription = (await self.session.execute(query)).scalar_one_or_none()
+        return subscription
+
+    async def get_subscription_by_customer_id(
+        self, customer_id: str
+    ) -> Optional[Subscription]:
+        query = select(Subscription).where(
+            Subscription.stripe_customer_id == customer_id
+        )
+        subscription = (await self.session.execute(query)).scalar_one_or_none()
+        return subscription
+
+    async def list_subscriptions_by_user_id(
+        self, user_id: str
+    ) -> List[Optional[Subscription]]:
+        query = select(Subscription).where(Subscription.user_id == user_id)
+        subscriptions = (await self.session.execute(query)).scalars().all()
+        return subscriptions
+
+    async def get_subscription_by_user_id(self, user_id: str) -> Optional[Subscription]:
```

Comment:
Hmmm I wonder if we should people to have only 1 Subscription object, and multiple payments